### PR TITLE
Deal with non-bug bugs currently in Raygun

### PR DESCRIPTION
**Pull request by DavidBoike**

### CODE REVIEW

File: src/AppCommon/Infra/InteractiveHttpAuth.cs [hunk 1]
```diff
@@ -51,44 +61,51 @@ public static async Task<Func<HttpClient>> CreateHttpClientFactory(Uri authUri,
...
-                }
-                catch (HttpRequestException x) when (response.StatusCode == HttpStatusCode.Unauthorized)
-                {
-                    socketHandler.Dispose();
-                    if (--maxTries <= 0)
+                    try
                     {
-                        throw new HaltException(HaltReason.Auth, "Unable to authenticate to " + uriPrefix, x);
-                    }
+                        _ = response.EnsureSuccessStatusCode();
```
Comment (awright18) [inline src/AppCommon/Infra/InteractiveHttpAuth.cs:70]: why do we need the discard at all here?

File: src/AppCommon/Infra/InteractiveHttpAuth.cs [hunk 2]
```diff
@@ -51,44 +61,51 @@ public static async Task<Func<HttpClient>> CreateHttpClientFactory(Uri authUri,
...
-                        Out.WriteLine($"  * {authHeader.Scheme} ({authHeader.Parameter})");
+                        return () => new HttpClient(socketHandler, disposeHandler: false);
                     }
-                    Out.WriteLine();
+                    catch (HttpRequestException x) when (response.StatusCode == HttpStatusCode.Unauthorized)
+                    {
+                        socketHandler.Dispose();
+                        if (--maxTries <= 0)
+                        {
+                            throw new HaltException(HaltReason.Auth, "Unable to authenticate to " + uriPrefix, x);
```
Comment (awright18) [inline src/AppCommon/Infra/InteractiveHttpAuth.cs:79]:
```suggestion
                            throw new HaltException(HaltReason.Auth, $"Unable to authenticate to {uriPrefix}", x);
```

File: src/AppCommon/Infra/Out.cs [hunk 3]
```diff
@@ -32,9 +32,16 @@ public static void Write(string text)
...
-        var result = Console.ReadLine();
-        output.AppendLine(result);
-        return result;
+        try
+        {
+            var result = Console.ReadLine();
+            output.AppendLine(result);
+            return result;
+        }
+        catch (UnauthorizedAccessException)
```
Comment (awright18) [inline src/AppCommon/Infra/Out.cs:41]: When would this throw? I don't understand, but I also haven't tried it.

Comment (DavidBoike) [reply src/AppCommon/Infra/Out.cs:41]: In an environment with a non-interactive shell, so stdin isn't available and so Console.ReadLine throws.

Comment (awright18) [reply src/AppCommon/Infra/Out.cs:41]: That makes sense! Thanks!

File: src/AppCommon/Infra/InteractiveHttpAuth.cs (hunk 1 above)
Comment (DavidBoike) [reply src/AppCommon/Infra/InteractiveHttpAuth.cs:70]: It's a fluent API, it returns itself.

**Review by awright18 — APPROVED** [review]

**Pull request merged**